### ISSUE TITLE
Feature request: mute

### ISSUE THREAD

**Issue by karlpokus** [sfarthin/intercept-stdout#8]
Thanks for sharing your module! I'm looking for a way to both intercept and mute the log. By commenting out [this line](https://github.com/sfarthin/intercept-stdout/blob/master/intercept-stdout.js#L26) I think it would work.

Would you be interested in a PR or should I fork?

**Comment by SimenB** [comment]
Not returning in the interceptor should work, shouldn't it?

**Comment by sfarthin** [comment]
I think SimenB is basically right, you can see the first example (output is muted) in the README compared to the 2nd one under the modification section (the 'intercepter' returns a manipulated string). Although, I think process.stdout.write is still being called regardless, but I am not sure this has an effect.

**Comment by karlpokus** [comment]
@SimenB Nor sure what you mean by _Not returning in the interceptor_. Commenting out `captured_text += txt` or removing the callback altogether does not work.

@sfarthin But the log is _not_ muted in the first example. The part below logs a and b.

```
var intercept = require("intercept-stdout"),
    captured_text = "";

var unhook_intercept = intercept(function(txt) {
    captured_text += txt;
});

console.log("a");
unhook_intercept(); // Let's stop capturing stdout.
console.log("b");
```

Nevermind. I made [something](https://github.com/karlpokus/console.mute) that mutes the log (although I think I lost the streaming aspect but that's fine by me).

**Comment by fedor** [comment]
@karlpokus correct. @sfarthin was talking about [_2nd_ example](https://github.com/fedor/intercept-stdout#modify) in the readme.

It's not obvious from the readme, but after studying tests I realised that to suppress output empty string should be returned (not `undefined`).

``` js
let unhook_intercept = intercept(function (line) {
    // ...
    return ''
})
```

@sfarthin could be useful to mention that not returning a string will result in print of the original string.

**Comment by sfarthin** [comment]
Sounds good, would you mind doing a PR?

**Comment by LaurentGoderre** [comment]
This sort of can be closed. The readme has the info but on the npm site, the changes to the readme are not there.
